src/ast/passes/semantic_analyser.cpp

```diff
@@ -1529,30 +1529,77 @@ Probe *SemanticAnalyser::get_probe_from_scope(Scope *scope,
 
 void SemanticAnalyser::visit(Map &map)
 {
-  MapKey key;
+  SizedType key = CreateNone();
+  bool is_tuple_ast = false;
+
+  if (map.key_expr) {
+    Visit(map.key_expr);
+    Expression *expr = map.key_expr;
+
+    if (expr->type.IsTupleTy()) {
+      // We can only do this expression and size type manipulation
+      // if we have access to the underlying expressions in the tuple ast
+      // e.g. @map[(1, 2)] vs $a = (1, 2) @map[$a]
+      auto tuple_ast = dynamic_cast<Tuple *>(expr);
+      if (tuple_ast) {
+        is_tuple_ast = true;
+        std::vector<SizedType> elements;
+        // NOLINTBEGIN(modernize-loop-convert)
+        for (unsigned int i = 0; i < tuple_ast->elems.size(); i++) {
+          // NOLINTEND(modernize-loop-convert)
```

Comment:
Why not accept the lint and use a range-based for-loop?

Reply:
Leftover from older code. I'll switch it to the range-based